include/lbann/execution_algorithms/ltfb/termination_criteria.hpp

```diff
@@ -0,0 +1,57 @@
+////////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2014-2021, Lawrence Livermore National Security, LLC.
+// Produced at the Lawrence Livermore National Laboratory.
+// Written by the LBANN Research Team (B. Van Essen, et al.) listed in
+// the CONTRIBUTORS file. <lbann-dev@llnl.gov>
+//
+// LLNL-CODE-697807.
+// All rights reserved.
+//
+// This file is part of LBANN: Livermore Big Artificial Neural Network
+// Toolkit. For details, see http://software.llnl.gov/LBANN or
+// https://github.com/LLNL/LBANN.
+//
+// Licensed under the Apache License, Version 2.0 (the "Licensee"); you
+// may not use this file except in compliance with the License.  You may
+// obtain a copy of the License at:
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the license.
+////////////////////////////////////////////////////////////////////////////////
+#ifndef LBANN_EXECUTION_ALGORITHMS_LTFB_TERMINATION_CRITERIA_HPP_INCLUDED
+#define LBANN_EXECUTION_ALGORITHMS_LTFB_TERMINATION_CRITERIA_HPP_INCLUDED
+
+#include "lbann/execution_algorithms/ltfb/execution_context.hpp"
+#include "lbann/execution_contexts/execution_context.hpp"
+
+namespace lbann {
+namespace ltfb {
+
+/** @class TerminationCriteria
+ *  @brief The stopping criteria for an LTFB-type algorithm
+ *
+ *  An object here needs to manage
+ */
+class TerminationCriteria final : public lbann::termination_criteria
+{
+public:
+  TerminationCriteria(size_t max_num_steps)
```

Comment:
`max_num_steps` is specified in the Python front-end.

A step of the `LTFB` training algorithm is defined as one round of training (which has its own stopping criteria!) followed by one call to the metalearning strategy. Thus, each trainer must specify the same value. It will handle a user's failure to do so by hanging some, possibly all, of the ranks (probably all). I can update this to assert that all trainers do the same number of steps.